src/main/kotlin/CheckersGame.kt

```diff
@@ -1,16 +1,18 @@
-class CheckersGame (private val model : CheckersModel, playerWhite : BasePlayer, playerBlack : BasePlayer) {
-    private val players = Array<BasePlayer>(2) { i -> if (i == 0) playerWhite else playerBlack}
+class CheckersGame (val model : CheckersModel, playerWhite : BasePlayer<CheckersModel>, playerBlack : BasePlayer<CheckersModel>) {
```

Comment:
I think that `BasePlayer<*>` will be ok here

Reply:
This class specializes exactly checkers game, so it is OK to use specialized template here